### ISSUE TITLE
Cannot emit events that have resolved relationships

### ISSUE THREAD

**Issue by sstone1** [hyperledger/composer#1113]
## Context

I cannot emit events that have resolved relationships. This makes it more painful then it needs to be, and is inconsistent with the behaviour of the asset and participant registry APIs:

```javascript
function onSampleTransaction(sampleTransaction) {
    var oldValue = sampleTransaction.asset.value;
    sampleTransaction.asset.value = sampleTransaction.newValue;
    return getAssetRegistry('org.acme.sample.SampleAsset')
        .then(function (assetRegistry) {
            return assetRegistry.update(sampleTransaction.asset);
        })
        .then(function () {
            var event = getFactory().newEvent('org.acme.sample', 'SampleEvent');
            // this does not work
            event.asset = sampleTransaction.asset;
            // meaning i have to do this
            event.assetId = sampleTransaction.asset.assetId;
            event.oldValue = oldValue;
            event.newValue = sampleTransaction.newValue;
            emit(event);
        });
}
```

## Expected Behavior

Resources that are in place of relationships, as is the case for a resolved relationship, should be silently serialized as a relationship without having to force the user to "unresolve" the relationship.

## Actual Behavior

Error is thrown due to the resolved relationship.

## Possible Fix

Use magical serializer option to handle this case.

## Steps to Reproduce

1.
2.
3.
4.

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [ ] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

**Comment by dselman** [comment]
This is also inconsistent with `post()` which supports this. 

```
options.convertResourcesToRelationships = true;
options.permitResourcesForRelationships = true;
const data = serializer.toJSON(typed, options);
```

**Comment by sstone1** [comment]
Fixed in #1114.
